Libplanet/Blockchain/Policies/BlockPolicy.cs

```diff
@@ -131,9 +131,20 @@ DateTimeOffset currentTime
         /// <inheritdoc />
         public int GetNextBlockDifficulty(IEnumerable<Block<T>> blocks)
         {
-            return ExpectDifficulties(blocks, yieldNext: true)
-                .First(t => t.Block == null)
-                .Difficulty;
+            int blockCount = blocks.Count();
+
+            if (blockCount <= 1)
+            {
+                return blockCount;
+            }
+
+            Block<T>[] prevBlocks = blocks.Skip(Math.Max(0, blockCount - 2))
```

Comment:
It seems to be better than before but still a loop with as much complexity as O(n). How about using the index in `BlockChain <T> .Store`?